### PR TITLE
Support stable MSC1772 spaces identifiers.

### PR DESCRIPTION
**Pull request by clokep**
Blocked on FCP matrix-org/matrix-doc#1772.

Also see matrix-org/matrix-js-sdk#1679, matrix-org/complement#107.

### CODE REVIEW

File: synapse/handlers/space_summary.py [hunk 1]
```diff
@@ -360,8 +360,9 @@ async def _get_child_events(self, room_id: str) -> Iterable[EventBase]:
             [
                 event_id
                 for key, event_id in current_state_ids.items()
-                # TODO: update once MSC1772 lands
+                # TODO: update once MSC1772 has been FCP for a period of time.
```
Comment (clokep) [inline synapse/handlers/space_summary.py:363]: It was suggested to do this when we stabilize the spaces summary MSC.

**Comment by clokep** [comment]
Note that this is ready, but I've left it as draft until the MSC merges.

**Review by erikjohnston — APPROVED** [review]
SEEMS LEGIT

**Review by erikjohnston — APPROVED** [review]

**Comment by clokep** [comment]
I pushed a follow-up to this in 70f0ffd2fcd815a065b4734ac606654a2e11dd28 which fixes the identifier for the room type, it's simply `type`, not `m.type`.

**Pull request merged**